discovery/src/server.rs

```diff
@@ -251,7 +252,7 @@ pub struct DiscoveryServer {
 impl DiscoveryServer {
     pub fn new(config: Config, port: &mut u16) -> Result<Self, Error> {
         let (discovery, cred_rx) = RequestHandler::new(config);
-        let address = SocketAddr::new(Ipv4Addr::UNSPECIFIED.into(), *port);
+        let address = SocketAddr::new(IpAddr::from_str("::0").unwrap(), *port);
```

Comment:
Please refer to: https://github.com/librespot-org/librespot/pull/1307#issuecomment-2308431046